### PR TITLE
Use Prometheus pushgateway for celery/pillowtop

### PR DESCRIPTION
**Pull request by sravfeyn**
@orangejenny

### CODE REVIEW

File: corehq/util/metrics/prometheus.py [hunk 1]
```diff
@@ -89,3 +97,16 @@ def _get_metric(self, metric_type, name, tags, documentation, **kwargs):
                 'expected_tags': metric._labelnames
             })
             raise
+
+    def _push_to_gateway(self):
+        registry = CollectorRegistry()
+        multiprocess.MultiProcessCollector(registry)
+        try:
+            push_to_gateway('localhost:9091', job='batch_mode', registry=registry)
```
Comment (orangejenny) [inline corehq/util/metrics/prometheus.py:105]: Is this supposed to be localhost?

Comment (sravfeyn) [reply corehq/util/metrics/prometheus.py:105]: Updated!

**Review by orangejenny — APPROVED** [review]

File: corehq/util/metrics/prometheus.py [hunk 2]
```diff
@@ -89,3 +92,13 @@ def _get_metric(self, metric_type, name, tags, documentation, **kwargs):
                 'expected_tags': metric._labelnames
             })
             raise
+
+    def _push_to_gateway(self):
+        registry = CollectorRegistry()
+        multiprocess.MultiProcessCollector(registry)
+        host = getattr(settings, 'PUSHGATEWAY_HOST')
```
[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/util/metrics/prometheus.py:100]: @sravfeyn @rohit-dimagi I don't see this being set anywhere in localsettings. Am I missing it or is that still outstanding?

[the remark this answers is not in the document]
Comment (rohit-dimagi) [reply corehq/util/metrics/prometheus.py:100]: > @sravfeyn @rohit-dimagi I don't see this being set anywhere in localsettings. Am I missing it or is that still outstanding?

Manually rolled out on two nodes celery and pillow
http://100.71.188.203:9091/#
http://100.71.188.180:9091/

**Pull request merged**